rocks-db/src/dump_client.rs

```diff
@@ -168,30 +171,30 @@ impl Storage {
             let tx_authority_cloned = tx_authority.clone();
             let metadata_key_set_cloned = metadata_key_set.clone();
             let authorities_key_set_cloned = authorities_key_set.clone();
-            iterator_tasks.spawn(async move {
-                Self::iterate_over_indexes(
-                    rx_cloned,
-                    shutdown_cloned,
-                    rx_indexes_cloned,
-                    tx_metadata_cloned,
-                    tx_creators_cloned,
-                    tx_assets_cloned,
-                    tx_authority_cloned,
-                    metadata_key_set_cloned,
-                    authorities_key_set_cloned,
-                )
-                .await
-            });
+            iterator_tasks.spawn(Self::iterate_over_indexes(
+                rx_cloned,
+                shutdown_cloned,
+                rx_indexes_cloned,
+                tx_metadata_cloned,
+                tx_creators_cloned,
+                tx_assets_cloned,
+                tx_authority_cloned,
+                metadata_key_set_cloned,
+                authorities_key_set_cloned,
+            ));
         }
 
+        // Iteration over `asset_static_data` column via CUSTOM iterator.
         let iter = self.asset_static_data.iter_start();
-        // collect batch of keys
         let mut batch = Vec::with_capacity(batch_size);
+        // Collect batch of keys.
         for k in iter
             .filter_map(|k| k.ok())
             .filter_map(|(key, _)| decode_pubkey(key.to_vec()).ok())
         {
             batch.push(k);
+
+            // When batch is filled, find `AssetIndex` and send it to `tx_indexes` channel.
             if batch.len() == batch_size {
                 let indexes = self
                     .get_asset_indexes(batch.as_ref())
```

Comment:
Cant we use std::mem::take(&mut batch). Then it will be no need to call batch.clear()